### PR TITLE
Update GHA artifact functions

### PR DESCRIPTION
**Pull request by paulb777**
Update update-artifact and download-artifact to v4

Update the `find` command in zip.yml for the slightly different resulting directory structure.

### CODE REVIEW

**Comment by ncooke3** [comment]
Does this have anything to do with why the nightlies did not upload the artifact for the job that was successful?

**Comment by paulb777** [comment]
I don't know, but thought I'd address the low-hanging fruit of updating these to eliminate the warnings.

File: .github/workflows/zip.yml [hunk 1]
```diff
@@ -422,7 +416,7 @@ jobs:
     - name: Move frameworks
       run: |
         mkdir -p "${HOME}"/ios_frameworks/
-        find "${GITHUB_WORKSPACE}/${FRAMEWORK_DIR}" -name "Firebase*latest.zip" -exec unzip -d "${HOME}"/ios_frameworks/ {} +
+        find "${GITHUB_WORKSPACE}" -name "Firebase*latest.zip" -exec unzip -d "${HOME}"/ios_frameworks/ {} +
```
Comment (paulb777) [inline .github/workflows/zip.yml:419]: @ncooke3 Would you confirm this structure change is ok for the release process?

Comment (ncooke3) [reply .github/workflows/zip.yml:419]: The directory structure of the the downloadable release artifacts looks the same.

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**